inference-engine/src/vpu/common/src/ngraph/operations/static_shape_non_maximum_suppression.cpp

```diff
@@ -14,65 +15,40 @@ constexpr NodeTypeInfo StaticShapeNonMaxSuppression::type_info;
 StaticShapeNonMaxSuppression::StaticShapeNonMaxSuppression(
         const Output<Node>& boxes,
         const Output<Node>& scores,
-        const Output<Node>& max_output_boxes_per_class,
-        const Output<Node>& iou_threshold,
-        const Output<Node>& score_threshold,
-        const StaticShapeNonMaxSuppression::BoxEncodingType box_encoding,
-        const bool sort_result_descending,
-        const element::Type& output_type)
-        : ngraph::op::v4::NonMaxSuppression(
-            boxes, scores, max_output_boxes_per_class, iou_threshold, score_threshold,
-            box_encoding, sort_result_descending, output_type) {
-    constructor_validate_and_infer_types();
-}
-
-StaticShapeNonMaxSuppression::StaticShapeNonMaxSuppression(
-        const Output<Node>& boxes,
-        const Output<Node>& scores,
-        const StaticShapeNonMaxSuppression::BoxEncodingType box_encoding,
-        const bool sort_result_descending,
-        const element::Type& output_type)
-        : ngraph::op::v4::NonMaxSuppression(boxes,
-              scores,
-              ngraph::opset3::Constant::create(element::i64, Shape{}, {0}),
-              ngraph::opset3::Constant::create(element::f32, Shape{}, {.0f}),
-              ngraph::opset3::Constant::create(element::f32, Shape{}, {.0f}),
-              box_encoding, sort_result_descending, output_type) {
+        const Output<Node>& maxOutputBoxesPerClass,
+        const Output<Node>& iouThreshold,
+        const Output<Node>& scoreThreshold,
+        const Output<Node>& softNmsSigma,
+        int centerPointBox,
+        const bool sortResultDescending,
+        const element::Type& outputType)
+        : ngraph::op::NonMaxSuppressionIE3(
+        boxes, scores, maxOutputBoxesPerClass, iouThreshold, scoreThreshold,
+        softNmsSigma, centerPointBox, sortResultDescending, outputType) {
     constructor_validate_and_infer_types();
 }
 
 std::shared_ptr<Node>
 StaticShapeNonMaxSuppression::clone_with_new_inputs(const OutputVector& new_args) const {
     check_new_args_count(this, new_args);
-    NODE_VALIDATION_CHECK(this,
-                          new_args.size() >= 2 && new_args.size() <= 5,
-                          "Number of inputs must be 2, 3, 4 or 5");
-
-    const auto& arg2 = new_args.size() > 2 ? new_args.at(2) : ngraph::opset3::Constant::create(element::i32, Shape{}, {0});
-    const auto& arg3 = new_args.size() > 3 ? new_args.at(3) : ngraph::opset3::Constant::create(element::f32, Shape{}, {.0f});
-    const auto& arg4 = new_args.size() > 4 ? new_args.at(4) : ngraph::opset3::Constant::create(element::f32, Shape{}, {.0f});
-
-    return std::make_shared<StaticShapeNonMaxSuppression>(
-          new_args.at(0),
-          new_args.at(1),
-          arg2,
-          arg3,
-          arg4,
-          m_box_encoding,
-          m_sort_result_descending,
-          m_output_type);
+    return std::make_shared<NonMaxSuppressionIE3>(new_args.at(0), new_args.at(1), new_args.at(2), new_args.at(3),
```

Comment:
Looks a bit strange that you call `NonMaxSuppressionIE3` constructor, not `StaticShapeNonMaxSuppression`

Reply:
Yeah, that's my mistake, fixed.